### PR TITLE
Network: support AzureFirewall update operation for rule collections

### PR DESCRIPTION
**Pull request by xccc-msft**

### CODE REVIEW

File: src/ResourceManagement/Network/AzureFirewallImpl.cs [hunk 1]
```diff
@@ -457,37 +462,17 @@ protected async override Task<AzureFirewallInner> GetInnerAsync(CancellationToke
...
-                Inner.NetworkRuleCollections.Add(networkRuleCollection);
-                networkRuleCollection = null;
-            }
-            else if (ipConfiguration != null)
-            {
-                Inner.IpConfigurations.Add(ipConfiguration);
-                ipConfiguration = null;
-            }
-            return this;
+            return Attach();
```
Comment (yungezz) [inline src/ResourceManagement/Network/AzureFirewallImpl.cs]: what's impact of remove existing appending rule action?

**Comment by yungezz** [comment]
HI @xccc-msft why this change is important in net fluent now? is same behavior in java fluent now?

**Comment by yaohaizh** [comment]
@yungezz this is requested by service team. On the other hand, the java(v1 & v2) lags behind for this support because Java doesn't support private DNS. Maybe, private DNS is a good service for you to ramp up

File: src/ResourceManagement/Network/AzureFirewallImpl.cs (hunk 1 above)
Comment (xccc-msft) [reply src/ResourceManagement/Network/AzureFirewallImpl.cs]: I just move this appending action into define method. This is to align with update operations which don't required appending action.

**Pull request merged**